From dcdd603828a30e763f0ee8ffc4a5d2d913115cef Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Thu, 10 Feb 2022 12:06:40 +0200
Subject: [PATCH] [SW-63244] gaudi2: fix sram start offset of zephyr preboot

The new zephyr preboot will start at offset 0x400 in SRAM
due to ARC limitation of exception vector (offset 0x0 is already
taken for img_size and fw_version).

Change-Id: I285525ab37e6ca640dbd8d8e3d9ececa69c5b4a2
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 835271f..114f120 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -66,7 +66,9 @@
 #define ARC_ICCM_BASE		(GEN_REGION_BASE + 0x7040000)
 #define ARC0_ICCM_BASE		(GEN_REGION_BASE + 0x7000000)
 #define SP_SRAM_PREBOOT_OFFSET	0x1000
-#define PREBOOT_START_OFFSET	0x8
+
+/* TODO: switch to start offset 0x0 once image descriptors are used */
+#define PREBOOT_START_OFFSET	0x400
 #define PPBOOT_START_OFFSET	0xC
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + PPBOOT_START_OFFSET)
 #define PREBOOT_TEXT_BASE	(SDRAM_BASE + PREBOOT_START_OFFSET)
